op-interop-mon/monitor/buffer.go

```diff
@@ -0,0 +1,52 @@
+package monitor
+
+import (
+	"container/ring"
+)
+
+// A RingBuffer is a generic [container/ring.Ring]
+// with convenient methods for adding and removing values.
+// The generic type T should be a pointer type or interface
+// because the default value of T will be returned by Pop and Peek if the value
+// is unset.
+type RingBuffer[T any] struct {
+	*ring.Ring
```

Comment:
It might be better to not embed this, and then we can prevent the caller from reaching past the API in this file and accessing the `any` Value of the `ring.Ring` type.

Reply:
Went ahead and did this [1bdb2cf](https://github.com/ethereum-optimism/optimism/pull/16382/commits/1bdb2cf36f3a2f76cf032912b033012328b4b87d)